src/overlays/actors/ovl_En_Dt/z_en_dt.c

```diff
@@ -15,7 +19,70 @@ void EnDt_Destroy(Actor* thisx, PlayState* play);
 void EnDt_Update(Actor* thisx, PlayState* play);
 void EnDt_Draw(Actor* thisx, PlayState* play);
 
-#if 0
+void EnDt_ChangeAnim(EnDt* this, s32 animIndex);
+void EnDt_SetupRegularState(EnDt* this, PlayState* play);
+void EnDt_SetupCutsceneNpcs(EnDt* this, PlayState* play);
+void EnDt_OfferRegularTalk(EnDt* this, PlayState* play);
+void EnDt_SetupMeetingCutscene(EnDt* this, PlayState* play);
+void EnDt_UpdateMeetingCutscene(EnDt* this, PlayState* play);
+void EnDt_FinishMeetingCutscene(EnDt* this, PlayState* play);
+void EnDt_OfferMeetingReward(EnDt* this, PlayState* play);
+void EnDt_OfferFinalNightTalk(EnDt* this, PlayState* play);
+void EnDt_TriggerFinalNightTalkEvent(EnDt* this, PlayState* play);
+void EnDt_TriggerMeetingRewardEvent(EnDt* this, PlayState* play);
+void EnDt_TriggerMeetingNotebookEvent(EnDt* this, PlayState* play);
+void EnDt_SetupFinalNightState(EnDt* this, PlayState* play);
+void EnDt_UpdateVisualState(EnDt* this);
+void EnDt_StartFinalNightTalk(EnDt* this);
+void EnDt_UpdateCutsceneFocusTarget(EnDt* this);
+
+typedef enum {
+    /* 0 */ EN_DT_NPC_STATE_DEFAULT,
+    /* 1 */ EN_DT_NPC_STATE_IDLE,
+    /* 2 */ EN_DT_NPC_STATE_VIEWING_MEETING,
+    /* 3 */ EN_DT_NPC_STATE_OFFERED_MEETING_REWARD,
+    /* 4 */ EN_DT_NPC_STATE_WAIT_FINAL_NIGHT_TALK,
+    /* 5 */ EN_DT_NPC_STATE_DONE_FINAL_NIGHT_TALK
+} EnDtNpcState;
+
+typedef enum {
+    /* 0 */ EN_DT_CS_STATE_NONE,
+    /* 1 */ EN_DT_CS_STATE_WAITING,
+    /* 2 */ EN_DT_CS_STATE_PLAYING
+} EnDtCutsceneState;
```

Comment:
It seems like the Baisen/Muto npcs also share this enum, so it might be a good candidate to extract into a shared header.